ouster-lidar/ouster-ros
# [NOETIC] Further reduce overhead in the ros driver

## Related Issues & PRs
- resolves #10 
- resolves #152 
- related #175 
- resolves #238 
- resolves #251 
- related #259
- resolves #304
- related #306 
- related #352

## Summary of Changes
- Add Jazzy to the build status
- Update to SDK client 0.11.1
- Switch to using LidarPacket and ImuPacket
  - Get's rid of the deprecation warnings
- Publish raw packets for the `os_driver`
- Add a flag to control the publish of organized point cloud vs non-organized
- Add a flag to control the publish of destaggerd point cloud vs staggered
- Add parameter to set the min/max lidar range on generated point clouds
- Add a new flag to `proc_mask` to optionally publish the raw packets when using os_driver

### TODO(s)
- ~optionally publish raw packets in os_driver~
- publicize the use of OpenMP in `cartesianT` & `point_cloud_compose` methods

## Validation
- Run the driver in all different modes of operations
- Verify that the newly added features:
  - When `destagger` is set to false verify the driver doesn't destagger the published point cloud
  - When `organized` is set to false verify that the published point cloud isn't 2D
  - Adjust `min_range` and `max_range` and observe the point cloud shrink accordingly.
- Verify that `/ouster/imu_packets`  and `/ouster/lidar_packet` appear when RAW flag is passed to `os_driver`
